[Xcode] Some targets accessing /System/Library/PrivateFrameworks through non-system include paths

emw-apple opened the pull request:
#### 01aabfa3d78335deafdd34e69d1d59d03d636f59
<pre>
[Xcode] Some targets accessing /System/Library/PrivateFrameworks through non-system include paths
<a href="https://bugs.webkit.org/show_bug.cgi?id=292451">https://bugs.webkit.org/show_bug.cgi?id=292451</a>
<a href="https://rdar.apple.com/150539568">rdar://150539568</a>

Reviewed by Alexey Proskuryakov.

Fix a few instances of configurations like this:

    FRAMEWORK_SEARCH_PATHS = $(inherited)
        $(WK_QUOTED_OVERRIDE_FRAMEWORKS_DIR)
        $(SDK_DIR)$(NORMAL_UMBRELLA_FRAMEWORKS_DIR);

AFAICT, the intention here is to give the compiler access staged WebKit
frameworks in downlevel builds, plus WebCore and WebKitLegacy inside the
WebKit.framework umbrella. But on embedded platforms, there is no WebKit
umbrella and NORMAL_UMBRELLA_FRAMEWORKS_DIR is just
/System/Library/PrivateFrameworks.

Regardless, even when these build settings refer only to WebKit
frameworks, they should be on the system path. Test tools should not
fail due to diagnostics in WebKit headers.

Along the way, move $(inherited) to the front of the list in these
settings, to align with all other projects.

* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBase.xcconfig: Also
  move SYSTEM_FRAMEWORK_SEARCH_PATHS to a better location in the file.
* Tools/WebKitTestRunner/Configurations/Base.xcconfig: Somehow this
  project was building without referencing WK_PRIVATE_SDK_DIR. Add it.

Canonical link: <a href="https://commits.webkit.org/294568@main">https://commits.webkit.org/294568@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d4b84f16b8c866d2ec881a814d5a03b0558025c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102329 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21998 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12312 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107490 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52965 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104368 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22306 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30504 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77862 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34854 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105335 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17274 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92373 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58200 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17108 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10401 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52323 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86934 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10472 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109865 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29461 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21724 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86842 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29825 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88569 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86431 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21988 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31250 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8964 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23703 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29389 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34685 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29200 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32523 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30759 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/TestWebKitAPI/Configurations/TestWebKitAPIBase.xcconfig:
```diff
@@ -26,9 +26,9 @@ GCC_ENABLE_OBJC_EXCEPTIONS = YES;
-FRAMEWORK_SEARCH_PATHS_cocoatouch = $(inherited);
+SYSTEM_FRAMEWORK_SEARCH_PATHS = $(SYSTEM_FRAMEWORK_SEARCH_PATHS_$(WK_COCOA_TOUCH));
+SYSTEM_FRAMEWORK_SEARCH_PATHS_ = $(inherited) $(SYSTEM_LIBRARY_DIR)/PrivateFrameworks $(SYSTEM_LIBRARY_DIR)/Frameworks/WebKit.framework/Versions/A/Frameworks;
+SYSTEM_FRAMEWORK_SEARCH_PATHS_cocoatouch = $(inherited);
```
We could perhaps have `$(inherited)` in the first `SYSTEM_FRAMEWORK_SEARCH_PATHS` line here, and avoid having to define `SYSTEM_FRAMEWORK_SEARCH_PATHS_cocoatouch` at all.

Also, it's concerning that some projects have `$(inherited)` first, and others last.

emw-apple committed:
[Xcode] Some targets accessing /System/Library/PrivateFrameworks through non-system include paths
https://bugs.webkit.org/show_bug.cgi?id=292451
rdar://150539568

Reviewed by Alexey Proskuryakov.

Fix a few instances of configurations like this:

    FRAMEWORK_SEARCH_PATHS = $(inherited)
        $(WK_QUOTED_OVERRIDE_FRAMEWORKS_DIR)
        $(SDK_DIR)$(NORMAL_UMBRELLA_FRAMEWORKS_DIR);

AFAICT, the intention here is to give the compiler access staged WebKit
frameworks in downlevel builds, plus WebCore and WebKitLegacy inside the
WebKit.framework umbrella. But on embedded platforms, there is no WebKit
umbrella and NORMAL_UMBRELLA_FRAMEWORKS_DIR is just
/System/Library/PrivateFrameworks.

Regardless, even when these build settings refer only to WebKit
frameworks, they should be on the system path. Test tools should not
fail due to diagnostics in WebKit headers.

Along the way, move $(inherited) to the front of the list in these
settings, to align with all other projects.

* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBase.xcconfig: Also
  move SYSTEM_FRAMEWORK_SEARCH_PATHS to a better location in the file.
* Tools/WebKitTestRunner/Configurations/Base.xcconfig: Somehow this
  project was building without referencing WK_PRIVATE_SDK_DIR. Add it.

Canonical link: https://commits.webkit.org/294568@main
